src/java.base/share/classes/java/util/Objects.java

```diff
@@ -367,7 +367,7 @@ public static boolean nonNull(@GuardSatisfied @Nullable Object obj) {
      */
     @CFComment({"nullness: TODO: treat like other nullness assertion methods in the Checker Framework."})
     @Pure
-    public static <T extends @NonNull Object> T requireNonNull(@GuardSatisfied T obj, @GuardSatisfied Supplier<String> messageSupplier) {
+    public static <T> T requireNonNull(@GuardSatisfied @NonNull T obj, @GuardSatisfied Supplier<String> messageSupplier) {
```

Comment:
Shouldn't the return type be `@NonNull T`?

Reply:
Yes, I've created a new pull request.  https://github.com/typetools/jdk/pull/38. I'll also update the JDK in checker-framework.